sos/report/plugins/ceph_mon.py

```diff
@@ -6,28 +6,72 @@
 #
 # See the LICENSE file in the source distribution for further information.
 
+import re
+
 from sos.report.plugins import Plugin, RedHatPlugin, UbuntuPlugin
 
 
 class CephMON(Plugin, RedHatPlugin, UbuntuPlugin):
+    """
+    This plugin serves to collect information on monitor nodes within a Ceph
+    cluster. It is designed to collect from several versions of Ceph, including
+    those versions that serve as the basis for RHCS 4 and RHCS 5.
+
+    Older versions of Ceph will have collections from locations such as
+    /var/log/ceph, whereas newer versions (as of this plugin's latest update)
+    will have collections from /var/log/ceph/<fsid>/. This plugin attempts to
+    account for this where possible across the host's filesystem.
+
+    Users may expect to see several collections twice - once in standard output
+    from the `ceph` command, and again in JSON format. The latter of which will
+    be placed in the `json_output/` subdirectory within this plugin's directory
+    in the report archive. These JSON formatted collections are intended to
+    aid in automated analysis.
+    """
 
     short_desc = 'CEPH mon'
 
     plugin_name = 'ceph_mon'
     profiles = ('storage', 'virt', 'container')
-    containers = ('ceph-(.*)?mon.*',)
-    files = ('/var/lib/ceph/mon/',)
+    # note: for RHCS 5 / Ceph v16 the containers serve as an enablement trigger
+    # but by default they are not capable of running various ceph commands in
+    # this plugin - the `ceph` binary is functional directly on the host
+    containers = ('ceph-(.*-)?mon.*',)
+    files = ('/var/lib/ceph/mon/', '/var/lib/ceph/*/mon*')
+    ceph_version = 0
```

Comment:
Is this required, as the `get_ceph_version` method returns 0 as a safe fallback? (other plugins' changes dont have this)

Reply:
It squelches a code analysis alert to have this here. Not strictly required, though.